sycl/test/plugins/enqueue-arg-order.cpp

```diff
@@ -0,0 +1,261 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple  %s -o %t.out
+// RUN: env SYCL_PI_TRACE=2 %GPU_RUN_PLACEHOLDER %t.out | FileCheck %s
+// RUN: env SYCL_PI_TRACE=2 %CPU_RUN_PLACEHOLDER %t.out | FileCheck %s
+// XFAIL: *
```

Comment:
Why XFAIL?

Reply:
Because there is a bug in how we pass arguments to 2D rect operations, and this test demonstrates it.  The bug isn't fixed yet. This test is marked XFAIL until the bug is fixed.    

If that's now how you want to proceed, let me know. I can remove the test from this PR and include it with the actual bug-fix.  I put it here, because its why these particular changes to SYCL_PI_TRACE are needed.